src/test/java/org/jvnet/hudson/test/RealJenkinsRuleTest.java

```diff
@@ -95,6 +95,16 @@ private static void _smokes(JenkinsRule r) throws Throwable {
         assertThat(assertThrows(RealJenkinsRule.StepException.class, () -> rr.then(RealJenkinsRuleTest::throwsException)).getMessage(),
             containsString("IllegalStateException: something is wrong"));
     }
+
+    @Test public void killedExternally() throws Throwable {
+        rr.startJenkins();
+        try {
+            rr.proc.destroy();
```

Comment:
Or I suppose it would work to `runRemotely` something that asynchronously calls `Jenkins.exit`?

Reply:
Yes, though exit code wouldn't be the same.